x-pack/otel/processor/beatprocessor/processor.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beatprocessor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors/add_host_metadata"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+)
+
+type beatProcessor struct {
+	logger     *zap.Logger
+	processors []beat.Processor
+}
+
+func newBeatProcessor(set processor.Settings, cfg *Config) (*beatProcessor, error) {
+	bp := &beatProcessor{
+		logger:     set.Logger,
+		processors: []beat.Processor{},
+	}
+
+	for _, processorConfig := range cfg.Processors {
+		processor, err := createProcessor(processorConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create processor: %w", err)
+		}
+		if processor != nil {
+			bp.processors = append(bp.processors, processor)
+		}
+	}
+
+	return bp, nil
+}
+
+func createProcessor(cfg map[string]any) (beat.Processor, error) {
+	if len(cfg) == 0 {
+		return nil, nil
+	}
+	if len(cfg) > 1 {
+		if len(cfg) < 10 {
+			configKeys := make([]string, 0, len(cfg))
+			for k := range cfg {
+				configKeys = append(configKeys, k)
+			}
+			return nil, fmt.Errorf("expected single processor name but got %v: %v", len(cfg), configKeys)
+		}
+		return nil, fmt.Errorf("expected single processor name but got %v", len(cfg))
+	}
+	for processorName, processorConfig := range cfg {
+		switch processorName {
+		case "add_host_metadata":
+			return createAddHostMetadataProcessor(processorConfig)
+		default:
+			return nil, fmt.Errorf("invalid processor name '%s'", processorName)
+		}
+	}
```

Comment:
[Suggestion]
The name of the processor in know, you can access the map directly and handle the processor not being there:


```suggestion
	pCfg, ok := cfg["add_host_metadata"]
	if !ok { [...] }
	[ ... ]
```